src/main/java/reactor/ipc/netty/http/client/HttpClientWSOperations.java

```diff
@@ -111,8 +111,10 @@ public void onInboundNext(ChannelHandlerContext ctx, Object msg) {
 					}
 				}
 				catch (WebSocketHandshakeException wshe) {
-					onInboundError(wshe);
-					return;
+					if (clientError) {
```

Comment:
not sure about that block. it seems to me this is a protocol error, and it should always be signalled through `onInboundError` (like [here](https://github.com/reactor/reactor-netty/blob/79b374758e1d37143623110c532208f703e0c830/src/main/java/reactor/ipc/netty/http/client/HttpClientOperations.java#L506) in `HttpClientOperations`). Furthermore, if `clientError` is false, this would lead to the exception being swallowed.